### PR TITLE
build p4-based sonic docker

### PR DESCRIPTION
**Pull request by lguohan**

### CODE REVIEW

File: .gitmodules [hunk 1]
```diff
@@ -13,3 +13,15 @@
 [submodule "sonic-swss"]
 	path = src/sonic-swss
 	url = https://github.com/Azure/sonic-swss
+[submodule "src/p4-switch/switch"]
+	path = src/p4-switch/switch
+	url = https://github.com/lguohan/switch
```
Comment (qiluo-msft) [inline .gitmodules]: It is possible to point to Azure organization repo? Pointing to personal repo is weird.

File: Makefile [hunk 2]
```diff
@@ -77,6 +77,19 @@ $(addprefix dockers/docker-syncd/deps/,syncd_1.0.0_amd64.deb libsairedis_1.0.0_a
 dockers/docker-syncd/deps/%.deb: src/%.deb
 	mkdir -p `dirname $@` && cp $< $(dir $@)
 
+## Rules: docker-sonic (p4)
+dockers/docker-sonic-p4/deps/swss_1.0.0_amd64.deb: src/p4/swss_1.0.0_amd64.deb
```
Comment (qiluo-msft) [inline Makefile]: suggest to merge 3 similar rules

File: .gitmodules (hunk 1 above)
Comment (lguohan) [inline .gitmodules]: right now, no.

File: Makefile (hunk 2 above)
Comment (lguohan) [inline Makefile]: merged

**Comment by stcheng** [comment]
Resolve the conflicting

**Pull request merged**